### ISSUE TITLE
WPF version not reported correctly when using ClickOnce

### ISSUE THREAD

**Issue by moshegutman** [microsoft/appcenter-sdk-dotnet#1088]
### **Description**

It looks like the WPF analytics module is using the Assembly file version for "version" when viewing data in App Center. I would like it to use the ClickOnce publish version, or better yet, let me set the version myself.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Publish any WPF app with ClickOnce
2. The version number displayed on the charts does not reflect the ClickOnce publish version number.

Copied from
https://github.com/microsoft/appcenter/issues/907

**Comment by AnastasiaKubova** [comment]
Hi @moshegutman! It looks like the feature request. Could you please clarify why change the version in the file which we use don't suitable for you?

**Comment by moshegutman** [comment]
The version that is used for display in App Center does not match the ClickOnce publish version highlighted below. I think this is a bug, not a feature request. Every time a new version of my app is published, the ClickOnce version is incremented, but App Center always displays the same version number. This is because App Center uses the Assembly File Version instead.

![clickonce](https://user-images.githubusercontent.com/3697619/63282785-33f27280-c275-11e9-8924-056117bc9fcb.png)

**Comment by lucahost** [comment]
Maybe this could have a relation to the appcenter distribute issue I experienced. 
I posted it to the dev-community https://developercommunity.visualstudio.com/idea/702256/error-deploying-msi-to-appcenter.html

**Comment by annakocheshkova** [comment]
Hi there. We use the version from `Package.appxmanifest`. I am not familiar with **ClickOnce** technology but it looks like a custom thing. I will mark this as feature request and file a ticket.
Thank you.

**Comment by moshegutman** [comment]
ClickOnce is not “custom” https://docs.microsoft.com/en-us/visualstudio/deployment/clickonce-security-and-deployment?view=vs-2019

It is a feature of Visual Studio. This is a bug, not a feature request.

**Comment by annakocheshkova** [comment]
@moshegutman we will look into this in any case and fix. Thanks for bringing it into attention!

**Comment by guperrot** [comment]
Rather than manually setting version, it should be better to try to use something like:
```
if (ApplicationDeployment.IsNetworkDeployed)
        return ApplicationDeployment.CurrentDeployment.CurrentVersion;
```

**Comment by moshegutman** [comment]
Oh I agree, but I don’t know how or where App center pulls version information from.

**Comment by guperrot** [comment]
Currently it uses AssemblyFileVersion which is not good indeed for ClickOnce. But we could fix the SDK to look for deployment version first and fall back to the other one.

**Comment by moshegutman** [comment]
Yes that would be great!

**Comment by Jamminroot** [comment]
Im closing the issue now since the fix has been merged.

**Comment by guperrot** [comment]
(we keep issues open until release for visibility)

**Comment by Jamminroot** [comment]
Included since [2.4.0-preview](https://github.com/microsoft/appcenter-sdk-dotnet/releases/tag/2.4.0-preview)
